spring-cloud-gcp-autoconfigure/pom.xml

```diff
@@ -83,11 +77,10 @@
             <artifactId>postgres-socket-factory</artifactId>
             <optional>true</optional>
         </dependency>
-
         <dependency>
             <groupId>org.postgresql</groupId>
             <artifactId>postgresql</artifactId>
-            <optional>true</optional>
+            <scope>test</scope>
```

Comment:
Why `test` vs `optional`. Also, if you're adding the `postgresql` driver, it seems that `mysql-connector-java` should also be in this POM. I think the reason it works without it is because we have the PostgreSQL test, but not a MySQL test in this module.

Reply:
This is only needed for the Postgre test.
The reason we don't need the MySQL one is because `com.google.cloud.sql.mysql-socket-factory` pulls it in transitively.
Same thing strangely doesn't happen with `com.google.cloud.sql.postgres-socket-factory`...

Reply:
That's not surprise. See its' POM:
```
   <dependencies>
        <dependency>
            <groupId>com.google.cloud.sql</groupId>
            <artifactId>jdbc-socket-factory-core</artifactId>
            <version>1.0.4</version>
        </dependency>
    </dependencies>
```
unlike that one for the `mysql-socket-factory`:
```
  <dependencies>
        <dependency>
            <groupId>mysql</groupId>
            <artifactId>mysql-connector-java</artifactId>
            <version>5.1.38</version>
        </dependency>
        <dependency>
            <groupId>com.google.cloud.sql</groupId>
            <artifactId>jdbc-socket-factory-core</artifactId>
            <version>1.0.4</version>
        </dependency>
    </dependencies>
```

Shouldn't it be fixed on the GCP level? Or let's just try to figure out what was the reason do not include `postgres` transitively.

Thanks

Reply:
I kind of assumed the Cloud SQL team had a good reason for having the MySQL dependency there..

In the end, I had to mark the PostgreSQL dependency as `<scope>compile</scope>` in order to adds its driver to the other `@ConditionalOnClass`.

What we can do is to exclude the MySQL dependency from the `mysql-socket-factory` import and then add it ourselves with `<scope>compile</scope>`..